index.js

```diff
@@ -1,110 +1,90 @@
-// ***************************************************************************
-// Iteration 1 - `for...of` loop
-// ***************************************************************************
+// Iteration 1
 
 const getFirstNames = arr => {
   const userFirstNames = [];
   for (let user of arr) {
-    // Your code goes here ...
+    userFirstNames.push(user.firstName)
   }
+  return userFirstNames;
 };
 
 getFirstNames(usersArray);
-// expected output:
-// [ 'Kirby', 'Tracie', 'Kendra', 'Kinney', 'Howard', 'Rachelle', 'Lizzie' ]
 
-// ***************************************************************************
-// Iteration 2 - `for...of` loop and ES6 string literals `${}`
-// ***************************************************************************
+// Iteration 2
 
 const getFullNames = arr => {
-  // Your code goes here ...
+  let userFullNames = [];
+  for (let user of arr) {
+    userFullNames.push(`${user.firstName} ${user.lastName}`)
+  }
+  return userFullNames
 };
 
 getFullNames(usersArray);
-// expected output:
-// [ 'Kirby Doyle', 'Tracie May', 'Kendra Hines', 'Kinney Howard',
-//   'Howard Gilmore', 'Rachelle Schneider', 'Lizzie Alford' ]
 
-// ***************************************************************************
-// Iteration 3 - ES6 destructuring , for of loop, object literal
-// ***************************************************************************
+// Iteration 3
 
 const getUsersCreditDetails = arr => {
-  // Your code goes here ...
+  let usersCreditDetails = [];
+  for (let user of arr) {
+    const {firstName, lastName, balance} = user;
+    const userDetails = {firstName, lastName, balance};
+    usersCreditDetails.push(userDetails)
+  }
+  return usersCreditDetails;
 };
 
 getUsersCreditDetails(usersArray);
-// expected output:
-// [ { firstName: 'Kirby', lastName: 'Doyle', balance: '$3,570.06' },
-// { firstName: 'Tracie', lastName: 'May', balance: '$1,547.73' },
-// { firstName: 'Kendra', lastName: 'Hines', balance: '$12,383.08' },
-// { firstName: 'Kinney', lastName: 'Howard', balance: '$3,207.06' },
-// { firstName: 'Howard', lastName: 'Gilmore', balance: '$21,307.75' },
-// { firstName: 'Rachelle', lastName: 'Schneider', balance: '$35,121.49' },
-// { firstName: 'Lizzie', lastName: 'Alford', balance: '$4,382.94' } ]
-
-// ***************************************************************************
-// Iteration 4 - practice `.filter()` method and how to return two elements
-// ***************************************************************************
+
+// Iteration 4
 
 const genderView = users => {
-  // Your code goes here ...
+  const femaleUsers = users.filter(user => user.gender === 'female').map((user) => {
+    const { firstName, lastName } = user;
+    return {firstName, lastName};
+  });
+  const maleUsers = users.filter(user => user.gender === 'male').map((user) => {
+    const { firstName, lastName } = user;
+    return {firstName, lastName};
+  });
```

Comment:
Beautiful! This a little bit more "meta" but if you had to do this anymore and repeat this kind of function you could have maybe also tried abstracting the filtering and mapping to another function, but this is exactly what was intended 💪